### PR TITLE
[FEATURE] Allocation by blobber

### PR DESCRIPTION
**Pull request by moldis**
This branch should be merged after ETH wallet features.

1. It's adding the possibility to get the cost of storage for listed blobbers only
2. Create allocation for listed blobbers only

All these features are mandatory for mobile clients

https://github.com/0chain/gosdk/pull/82

### CODE REVIEW

**Comment by moldis** [comment]
@NoSkillGuy Pls confirm or leave comments here, otherwise I will merge it as need this feature for Android/IOS

File: zboxcore/sdk/allocation.go [hunk 1]
```diff
@@ -1331,12 +1331,12 @@ func (a *Allocation) RemoveCollaborator(filePath, collaboratorID string) error {
...
 }
 
-func (a *Allocation) GetMaxWriteRead() (maxW float64, maxR float64, err error) {
+func (a *Allocation) GetMaxWriteReadFromBlobbers(blobbers []*BlobberAllocation) (maxW float64, maxR float64, err error) {
 	if !a.isInitialized() {
 		return 0, 0, notInitialized
 	}
 
-	blobbersCopy := a.BlobberDetails
+	blobbersCopy := blobbers
```
Comment (cnlangzi) [inline zboxcore/sdk/allocation.go:1339]: why we need a `blobbersCopy`?

File: zboxcore/sdk/allocation.go [hunk 2]
```diff
@@ -1377,6 +1381,22 @@ func (a *Allocation) GetMinWriteRead() (minW float64, minR float64, err error) {
 	return minWritePrice, minReadPrice, nil
 }
 
+func (a *Allocation) GetMaxStorageCostFromBlobbers(size int64, blobbers []*BlobberAllocation) (float64, error) {
+	var cost common.Balance // total price for size / duration
+
+	for _, d := range blobbers {
+		fmt.Printf("write price for blobber %f datashards %d parity %d\n",
```
Comment (cnlangzi) [inline zboxcore/sdk/allocation.go:1388]: please don't use fmt.Printf to print log. It is impossible to turn off in config file.

**Review by cnlangzi — CHANGES_REQUESTED** [review]

**Review by cnlangzi — APPROVED** [review]
LGTM

**Pull request merged**